tools/jenkins/Jenkinsfile-PR

```diff
@@ -0,0 +1,130 @@
+	stages {
+		stage('parallel') {
+			failFast false
+			parallel {
+				stage('pc-tests') {
+					agent { label 'ooc' }
+					steps {
+						sh 'echo ">>>> Agent: " `hostname`'
+						timeout(unit: 'MINUTES', time: 20) {
+							checkout([$class: 'GitSCM',
+							branches: [[name: '*/develop']],
+							doGenerateSubmoduleConfigurations: false,
+							extensions: [[$class: 'RelativeTargetDirectory',
+							relativeTargetDir: 'vidhance-sdk']],
+							submoduleCfg: [],
+							userRemoteConfigs: [[credentialsId: 'vidhance-admin-login', url: 'https://github.com/vidhance/vidhance-sdk']]
+							])
+						}
+
+						sh 'cp -r /home/print/media/ vidhance-sdk/media/'
+						sh 'magic'
+
+						sh 'cd vidhance-sdk && ./tools/ff_combinations.sh'
+						sh 'cd vidhance-sdk && ./build.sh -DdebugGL +-Werror'
+						sh 'cd vidhance-sdk && ./main ~/dev/media/videos/sideways1.mp4 -b 20'
+						sh 'cd vidhance-sdk && ./tools/memplay.sh ~/dev/media/videos/sideways1.mp4 -b 10'
+						sh 'cd vidhance-sdk && cat memcheck.valgrindlog'
+
+						sh 'cp /home/print/versioned/util/print/settings/all.ooc vidhance-sdk/source/vidhance/sdk/pc/Settings.ooc'
+						sh 'cd vidhance-sdk && rm -f main'
+						sh './test.sh all -x -DdebugGL -Dsafe +-Werror'
+						sh './test.sh all -DdebugGL +-Werror'
+						// DSafe
+						sh 'cd vidhance-sdk && ./test.sh all -x -Dsafe -DdebugGL +-Werror'
+						sh 'cd vidhance-sdk && ./build.sh -Dsafe -DdebugGL +-Werror'
+						sh 'cd vidhance-sdk && ./main ~/dev/media/videos/sideways1.mp4 -w -b 10'
+						sh 'cd vidhance-sdk && ./main ~/dev/media/videos/sideways1.mp4 -b 20'
+						// API
+						sh 'cd vidhance-sdk && rm -f main'
+						sh 'cd vidhance-sdk && rock -x'
+						sh 'cd vidhance-sdk && ./test.sh api +-Wall -DdebugGL +-Werror'
+						sh 'cd vidhance-sdk && rm -f main'
+						// GPU
+						sh 'cd vidhance-sdk && ./test.sh all -DdebugGL -x +-Werror'
+						sh 'cd vidhance-sdk && ./build.sh -DdebugGL +-Werror'
+						sh 'cd vidhance-sdk && ./main ~/dev/media/videos/sideways1.mp4 -b 20'
+						sh 'cd vidhance-sdk && ./tools/memplay.sh ~/dev/media/videos/sideways1.mp4 -b 10'
+						sh 'cd vidhance-sdk && cat memcheck.valgrindlog'
+
+						sh 'cp /home/print/versioned/util/print/settings/stabilize-gyro.ooc vidhance-sdk/source/vidhance/sdk/pc/Settings.ooc'
+						sh 'cd vidhance-sdk && ./build.sh -x -DdebugGL +-Werror'
+						sh 'cd vidhance-sdk && ./main ~/dev/media/videos/sideways1.mp4 -b 20'
+
+						sh 'cp /home/print/versioned/util/print/settings/autozoom-on.ooc vidhance-sdk/source/vidhance/sdk/pc/Settings.ooc'
+						sh 'cd vidhance-sdk && ./build.sh -x -DdebugGL +-Werror'
+						sh 'cd vidhance-sdk && ./main ~/dev/media/videos/sideways1.mp4 -b 30'
+					}
+					post {
+						always {
+							sh 'rm -r vidhance-sdk/media/'
+						}
+					}
+				}
+				stage('android-tests') {
+					agent { label 'android-7.1.2' }
+					steps {
+						sh 'echo ">>>> Agent: " `hostname`'
+						timeout(unit: 'MINUTES', time: 20) {
```

Comment:
Group the checkout together with the steps that are in that repo.
Besides, the `timeout` is superflouous.